pkg/client/cache/eventqueue.go

```diff
@@ -36,12 +36,13 @@ import (
 // watch.Deleted event (though this state is delivered by the watch API itself, it is not passed on
 // to the reflector Store).
 type EventQueue struct {
-	lock   sync.RWMutex
-	cond   sync.Cond
-	store  kcache.Store
-	keyFn  kcache.KeyFunc
-	events map[string]watch.EventType
-	queue  []string
+	lock           sync.RWMutex
+	cond           sync.Cond
+	store          kcache.Store
+	keyFn          kcache.KeyFunc
+	events         map[string]watch.EventType
+	queue          []string
+	lastReplaceKey string
```

Comment:
Ok, makes sense.  There is definitely some specialized knowledge of reflectors and stores that is required to understand this solution.